packages/zoe/zoe.js

```diff
@@ -182,39 +276,43 @@ const makeZoe = async (additionalEndowments = {}) => {
        * they are getting. Note: if information can be derived in
        * queries based on other information, we choose to omit it. For
        * instance, `installationHandle` can be derived from
-       * `instanceId` and is omitted even though it is useful.
+       * `instanceHandle` and is omitted even though it is useful.
        * @param  {object} contractDefinedExtent - an object of
        * information to include in the extent, as defined by the smart
        * contract
-       * @param  {object} useObj - an object defined by the smart
+       * @param  {object} seat - an object defined by the smart
        * contract that is the use right associated with the invite. In
        * other words, buying the invite is buying the right to call
        * methods on this object.
        */
-      makeInvite: (contractDefinedExtent, useObj) => {
-        const inviteExtent = harden({
-          ...contractDefinedExtent,
-          handle: harden({}),
-          instanceHandle,
-        });
-        const invitePurseP = inviteMint.mint(inviteExtent);
-        inviteAddUseObj(inviteExtent.handle, useObj);
-        const invitePaymentP = invitePurseP.withdrawAll();
-        return invitePaymentP;
-      },
-
-      /** read-only, side-effect-free access below this line */
-      getStatusFor: readOnlyState.getStatusFor,
-      makeEmptyExtents: () =>
-        makeEmptyExtents(
-          readOnlyState.getExtentOpsArrayForInstanceHandle(instanceHandle),
-        ),
-      getExtentOpsArray: () =>
-        readOnlyState.getExtentOpsArrayForInstanceHandle(instanceHandle),
-      getLabels: () => readOnlyState.getLabelsForInstanceHandle(instanceHandle),
-      getExtentsFor: readOnlyState.getExtentsFor,
-      getPayoutRulesFor: readOnlyState.getPayoutRulesFor,
+      makeInvite: (contractDefinedExtent, seat) =>
+        makeInvite(instanceHandle, seat, contractDefinedExtent),
       getInviteAssay: () => inviteAssay,
+
+      // To be used by contracts in the near future.
```

Comment:
I think there are more decolumnization opportunities here that will make some of these go away. But such further progress should happen after this PR, since it is more entangled with other upcoming changes.

Reply:
Agreed